server/s3_bucket_metadata.cc

```diff
@@ -30,7 +30,8 @@
 #include "s3_m0_uint128_helper.h"
 
 S3BucketMetadata::S3BucketMetadata(
-    std::shared_ptr<S3RequestObject> req, std::shared_ptr<MotrAPI> motr_api,
+    std::shared_ptr<S3RequestObject> req, std::string bucket,
```

Comment:
I don't understand why we pass 'std::string bucket' to S3BucketMetadata's constructor.
You can always get bucket name from RequestObject class. 
As CopyObject involves source and destination buckets, and both these are part of the request, these bucket names should be part of "RequestObject" or "S3RequestObject" class. You may add method (get/set) in S3RequestObject class to encapsulate the second bucket name.

Reply:
Yes Datta you are right, I could have updated S3RequestObject class to have 2 new member variables - source_bucket and source object.
And I have no strong counter point as to why I have not done this.

My thought process was that as this is Bucket metadata class, the constructor should also be able to load metadata of a given bucket, instead of just relying on the S3RequestObject class' encapsulated bucket.

May be in future, if there arises a need for some API (new or existing) to load metadata of a (or more) given bucket(s) (other than in S3RequestObject class), then we can easily load its metadata with current changes. 
I could have written one more constructor to add this functionality, but that could have lead to code duplication.

1 more point in favor of my argument is - that for authorization, these changes would reduce code changes for authserver (after talks with @abhilekh268 )

Reply:
After discussion with Datta, We have decided to not add member variables to S3RequestObject, as is suggested above.

We just want opinion from @knrajnambiar76, as to change the signature of the current constructor (as is done in above changes) or add new constructor, which would take bucket name as parameter, instead of S3RequestObject.

Can you comment @knrajnambiar76 here your opinion ?

Reply:
2nd option suggested by @DPG17 , to make temporary copy of S3RequestObject class in the Action class, and set the bucket name to source bucket name. This way we do not need to change signature of bucket metadata class.

Reply:
I added a new constructor, which accepts bucket name, along with other parameters.